packages/location_permissions/lib/src/location_permissions.dart

```diff
@@ -10,23 +10,23 @@ class LocationPermissions {
     if (_instance == null) {
       const MethodChannel methodChannel =
           MethodChannel('com.baseflow.flutter/location_permissions');
-      final EventChannel eventChannel = Platform.isAndroid
+      final EventChannel? eventChannel = Platform.isAndroid
           ? const EventChannel(
               'com.baseflow.flutter/location_permissions_events')
           : null;
 
       _instance = LocationPermissions.private(methodChannel, eventChannel);
     }
-    return _instance;
+    return _instance!;
   }
 
   @visibleForTesting
   LocationPermissions.private(this._methodChannel, this._eventChannel);
 
-  static LocationPermissions _instance;
+  static LocationPermissions? _instance;
```

Comment:
This can be decorated with the `late` initialiser, since the `_instance` variable will always be initialised through the factory constructor. 

```suggestion
  late static LocationPermissions _instance;
```

Reply:
How would I modify the singleton pattern in the factory when I change this? I would then not be able to do the null check anymore. I think this should stay optional because you can only access the instance via the factory method and this is only used internally to determine whether there already is a singleton instance.

Reply:
Yes you are right, completely overlooked this. Thanks for waking me up.